control-plane/pkg/contract/serde.go

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2020 The Knative Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package contract
+
+import (
+	"google.golang.org/protobuf/encoding/protojson"
+	"google.golang.org/protobuf/proto"
+)
+
+type format string
+
+const (
+	Protobuf format = "protobuf"
+	Json     format = "json"
+)
+
+type Serializer interface {
+	Serialize(ct *Contract) ([]byte, error)
+}
+
+type Deserializer interface {
+	Deserialize([]byte) (*Contract, error)
+}
```

Comment:
IS there a similar concept in other "golang" toolkits for Kafka?

(besides the java SerDes?)

Reply:
I don't think so because they need to be generic over a type `T`.